modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -922,6 +922,146 @@ GAPI_FLUID_KERNEL(GFluidNot, cv::gapi::core::GNot, false)
     }
 };
 
+//--------------------------------------
+//
+// Fluid math kernels: bitwise with Scalar
+//
+//--------------------------------------
+
+static void checkScalarForBitwise(const cv::Scalar &_scalar, const int scalarI[4])
+{
+    if (!((_scalar[0] == scalarI[0]) && (_scalar[1] == scalarI[1]) &&
+          (_scalar[2] == scalarI[2]) && (_scalar[3] == scalarI[3])))
+    {
+        CV_Error(cv::Error::StsBadArg, "Bitwise operations make sense with integral types only");
+    }
+}
+
+template<typename DST>
+static inline DST bw_andS(DST x, int y)
+{
+    return x & saturate<DST>(y);
+}
+
+template<typename DST>
+static inline DST bw_orS(DST x, int y)
+{
+    return x | saturate<DST>(y);
+}
+
+template<typename DST>
+static inline DST bw_xorS(DST x, int y)
+{
+    return x ^ saturate<DST>(y);
+}
+
+// manually unroll the inner cycle by channels
+// (reuse arithmetics function above of the same purpose)
+template<typename DST, typename FUNC>
+static inline void run_bitwise_s(DST out[], const DST in[], int width, int chan,
+                                 const int scalar[4], FUNC func)
+{
+    run_arithm_s(out, in, width, chan, scalar, func);
+}
+
+template<typename DST, typename SRC>
+static void run_bitwise_s(Buffer &dst, const View &src, const int scalar[4], Bitwise bitwise)
```

Comment:
Maybe it's better to rename `bitwise` to `bitwise_op`/`op`/etc?

Reply:
I don't think so: there are functions for bitwise operations for two Mats right above, and they are called `run_bitwise`. And, like with arithmetic functions which are changed from `run_arithm` to `run_arithm_s` for Scalar, I did the same for new functions.

But maybe I chose not so good names for little functions which actually do the operations - `bw_andS`, `bw_orS`, `bw_xorS`. What is your opinion about them?

Reply:
You get me wrong:) I'm talking about the argument `Bitwise bitwise` here

Reply:
aaa OK I see now :)
Nevertheless, in a function for two-Mats bitwise operation above it is `Bitwise bitwise` too, so..
It was done, I assume, for this string to be readable:
```
switch (bitwise)
```
But, if there wasn't a similar arg above, I'd totally agree with you

Reply:
Still you can change the names above

Reply:
Yes, of course. so if you think it's better to change, I'll do that

Reply:
Yes, please

Reply:
Renamed